### PR TITLE
Fixes #17303 - correct position of validation msg for parameters

### PR DESCRIPTION
**Pull request by kgaikwad**
Instead of showing all validation error messages on top, this commit will show each validation error message on the top of respective parameter.

### CODE REVIEW

**Comment by tbrisker** [comment]
Ok to test

**Comment by orrabin** [comment]
The error is now in red and above the parameter that is actually causing it as requested:
image

Maybe stretch the error on the whole row so it doesn't push down each row so much?

Also, please change the commit message to the current status and not the bug, for example it can be: `Fixes #17303 - correct position of validation msg for parameters`

**Comment by kgaikwad** [comment]
@orrabin, I have made suggested code changes. Added "colspan" attribute to table row cell.
Please have a look.

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -154,7 +154,10 @@ function check_caps_lock(key, e) {
 
 function remove_fields(link) {
   $(link).prev("input[type=hidden]").val("1");
-  $(link).closest(".fields").hide();
+  var $field_row = $(link).closest(".fields"),
+  $error_block = $field_row.prev("tr.error-msg-block");
+  if ($error_block) { $error_block.hide(); }
```
Comment (orrabin) [inline app/assets/javascripts/application.js]: Nitpick: no need for the if, calling hide on an empty error_block won't cause an error.

**Comment by orrabin** [comment]
@kgaikwad I tested again and it looks much better when stretched, thanks for the change.
Comparing this to the other errors in tables I noticed 2 things:
1) We don't remove the error when clicking remove :+1:  for adding that, maybe a separate pr can remove it in other places.
2) The rest of the tables put the error message under the row that causes it and not above.
Can you try moving the error under the row?

**Comment by kgaikwad** [comment]
@orrabin , done with the changes.
As I have removed top border of all rows for better UI with error indication, added little padding on top of each field row and removed the top margin of error row.

**Comment by kgaikwad** [comment]
[Test]

**Comment by orrabin** [comment]
[test foreman]

**Comment by orrabin** [comment]
@Rohoover this pr fixes the location of validation errors in the parameters table, they used to be above the table:
image

After this pr they are located correctly in the table:
image

but the table doesn't have borders to avoid borders between the parameter and the error (avoid the table below):
image

is it better to keep the table without any borders? with the borders always (even when there is an error)? should it look different?

**Comment by Rohoover** [comment]
@orrabin

I don't think the table borders (lines) are necessary as the fields themselves clearly define the rows. If borders are required, I would not put the error in its own row with borders - that visually separates the error from the field and could be confusing. 

Thanks!

**Comment by orrabin** [comment]
Thanks @kgaikwad!

**Pull request merged**